util/src/main/java/org/datacommons/util/SummaryReportGenerator.java

```diff
@@ -0,0 +1,99 @@
+package org.datacommons.util;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import org.datacommons.proto.Debug;
+
+public class SummaryReportGenerator {
+
+  // An object to save the information about a stat var. This contains all the necessary getters to
+  // access the information in this object from SummaryReport.ftl
+  public static final class StatVarSummary {
+    int numObservations = 0;
+    Set<String> places = new HashSet<>();
+    Set<String> mMethods = new HashSet<>();
+    Set<String> units = new HashSet<>();
+    Set<String> scalingFactors = new HashSet<>();
+    // This set will not be populated when StatVarSummary is generated by PlaceSeriesSummary
+    TreeSet<String> dates = new TreeSet<>();
```

Comment:
Given this is updated in the fast-path, if we only need the ordering at rendering time, perhaps use Set<> and sort? (https://stackoverflow.com/a/4464394)

Reply:
done